domain-model-assistant/Assets/Components/Scripts/CompartmentedRectangle.cs

```diff
@@ -64,20 +61,20 @@ public void CreateHeader()
     {
         var header = GameObject.Instantiate(textbox, this.transform);
         //vector position will need to be obtained from transform of gameobject in the future
-        header.transform.position = this.transform.position + new Vector3(0, 70, 0);
+        header.transform.position = this.transform.position + new Vector3(headerOffsetX, headerOffsetY, 0);
         AddHeader(header);
     }
 
     public void CreateSection()
     {
         Vector3 oldPosition = this.transform.position + new Vector3(0, 18, 0);
-        for (int i = 0; i < 2; i++)/*this loop assumes that class always has 2 sections*/
+        for (int i = 0; i < 2; i++)/*loop for a class with 2 sections*/
         {
             var sect = GameObject.Instantiate(section, this.transform);
-            sect.transform.position = oldPosition + new Vector3(0, -71, 0) * sections.Count;
+            sect.transform.position = oldPosition + new Vector3(0, sectionOffsetY, 0) * sections.Count;
             AddSection(sect);
-            _diagram.AddAttributes(GetSection(i), i);//add atrributes to section
         }
+        _diagram.AddAttributesToSection(GetSection(0));//add atrributes to first section
```

Comment:
Why do we need to call the diagram to do this?

Reply:
because this method was created in the `Diagram` class